### PR TITLE
IDENTITY-6076: Fix setting cookie expiration, domains and paths previ…

### PR DESCRIPTION
**Pull request by emswbandara**
…ously set to the cookie not getting set at logout

### CODE REVIEW

File: components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java [hunk 1]
```diff
@@ -850,26 +850,35 @@ private void storeTokenIdCookie(String sessionId, HttpServletRequest req, HttpSe
...
+
         if (samlssoTokenIdCookieConfig != null) {
-            samlssoTokenIdCookie.setMaxAge(samlssoTokenIdCookieConfig.getMaxAge() > 0 ?
-                    samlssoTokenIdCookieConfig.getMaxAge() :
-                    defaultMaxAge);
-            samlssoTokenIdCookie.setDomain(samlssoTokenIdCookieConfig.getDomain());
-        } else {
-            samlssoTokenIdCookie.setMaxAge(defaultMaxAge);
+            int age = samlssoTokenIdCookieConfig.getMaxAge() > 0 ? samlssoTokenIdCookieConfig.getMaxAge() :
+                    defaultMaxAge;
```
Comment (thanujalk) [inline components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java]: remove inline if

File: components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java [hunk 2]
```diff
@@ -850,26 +850,35 @@ private void storeTokenIdCookie(String sessionId, HttpServletRequest req, HttpSe
...
                         log.debug("SSO tokenId Cookie is removed");
                     }
+                    cookie.setHttpOnly(true);
+                    cookie.setSecure(true);
+                    cookie.setPath("/");
+
+                    if (samlssoTokenIdCookieConfig != null) {
+                        updateSAMLSSOIdCookieConfig(cookie, samlssoTokenIdCookieConfig, 0);
+                    }
                     cookie.setMaxAge(0);
```
Comment (thanujalk) [inline components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java]: unnecessary setMaxAge

File: components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java (hunk 1 above)
Comment (thanujalk) [inline components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java]: In this case, we give priority to max age value configured through identity.xml over session idle time out configured for that tenant. Is this correct?

Comment (emswbandara) [inline components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java]: fixed with 36e5ee9

**Pull request merged**